src/redux/books/books.js

```diff
@@ -1,57 +1,87 @@
-import { v4 as uuidv4 } from 'uuid';
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const POST_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/Tgl0nQKaAYBzzdbCfHYY/books';
 
 const ADD_BOOK = 'bookstore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
+const FETCH_BOOKS = 'FETCH_BOOKS';
 
-const initState = [
-  {
-    id: uuidv4(),
-    title: 'Rethinking Productivity in Software Engineering',
-    author: 'Caitlin Sadowski, Thomas Zimmermann',
-  },
-  {
-    id: uuidv4(),
-    title: 'Pro Git',
-    author: 'Scott Chacon and Ben Straub',
-  },
-  {
-    id: uuidv4(),
-    title: "You Don't Know JS Yet",
-    author: 'Kyle Simpson',
-  },
-  {
-    id: uuidv4(),
-    title: 'Learning JavaScript Design Patterns',
-    author: 'Addy Osmani',
-  },
-  {
-    id: uuidv4(),
-    title: 'Understanding ECMAScript 6',
-    author: 'Nicholas C. Zakas',
+const initState = [];
+
+export const fetchBookApi = () => async (dispatch) => {
+  const response = await axios.get(POST_URL);
+  const data = await response.data;
+
+  const books = Object.entries(data).map(([key, value]) => {
+    const { title, category, author } = value[0];
+    return {
+      itemId: key,
+      title,
+      category,
+      author,
+    };
+  });
+  if (books) {
+    dispatch({
+      type: FETCH_BOOKS,
+      payload: books,
+    });
+  }
+};
+
+export const addNewBook = createAsyncThunk(
+  'books/addNewBook',
+  async (initBooks) => {
+    try {
+      const response = await axios.post(POST_URL, initBooks);
+      return response.data;
+    } catch (error) {
+      return error.message;
+    }
   },
-];
+);
+
+export const deleteBook = (itemId) => async (dispatch) => {
+  const response = await axios.delete(`${POST_URL}/${itemId}`);
+  if (response.status === 201) {
+    dispatch({
+      type: REMOVE_BOOK,
+      itemId,
+    });
+  }
+};
```

Comment:
- [ ] When I add or remove a book I have to refresh the page to see the changes, Please fix this issue and users should be able to see the changes without refreshing the page.
  - This is hapning because you are not updating the local state on add or remove the book. Inside the API functions call the actions so It will update the local state.